test/e2e/kubebuildercontext.go

```diff
@@ -0,0 +1,169 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+)
+
+type CmdContext struct {
+	// environment variables in k=v format.
+	Env []string
+	Dir string
+}
+
+// KbContext specified to run e2e tests
+type KbContext struct {
```

Comment:
May be we can rename this to `TestContext` ?

Reply:
Renamed it to `KBTestContext` per https://github.com/kubernetes-sigs/kubebuilder/pull/767#discussion_r291772099